docs/pages/2020_Intro_Week/sections/workflows_basic.rst

```diff
@@ -1,6 +1,637 @@
+For this purpose, we have provided the ``plot_eos`` script in the ``common_wf.py`` file that takes the PK of the work function as an input and plots the equation of state:
+
+.. code-block:: ipython
+
+    In [7]: from common_wf import plot_eos
+       ...: plot_eos(359)
+
+Submitting the workflow: Workchains
+-----------------------------------
+
+Similar to the simple arithmetic work function above, running the ``eos_wf`` work function means that the Python interpreter will be blocked during the whole workflow.
+In this case, this will take the time required to launch the calculations, the actual time needed by Quantum ESPRESSO to perform the calculation and the time taken to retrieve the results.
+Perhaps you killed the calculation and you experienced the unpleasant consequences: intermediate calculation results are potentially lost and it is extremely difficult to restart a workflow from the exact place where it stopped.
+
+Clearly, when writing workflows that involve the use of an *ab initio* code like Quantum ESPRESSO, it is better to use a work chain.
+Below you can find the basic rules that allow you to convert your workfunction-based script to a workchain-based one and a snippet example focusing on the code used to perform the calculation of an equation of state.
+
+.. include:: include/snippets/workflows_eos_workchain.py
+    :code: python
+
+.. warning::
+
+    WorkChains need to be defined in a **separate file** from the script used to run them.
+    E.g. save your WorkChain in ``workchains.py`` and use ``from workchains import MyWorkChain`` to import it in your script.
+
+-   Instead of using decorated functions you need to define a class, inheriting from a prototype class called ``WorkChain`` that is provided by AiiDA in the ``aiida.engine`` module.
+
+-   Within your class you need to implement a ``define`` classmethod that always takes ``cls`` and ``spec`` as inputs.
+    Here you specify the main information on the workchain, in particular:
+
+    -   The *inputs* that the workchain expects.
+        This is obtained by means of the ``spec.input()`` method, which provides as the key feature the automatic validation of the input types via the ``valid_type`` argument.
+        The same holds true for outputs, as you can use the ``spec.output()`` method to state what output types are expected to be returned by the workchain.
+
+    -   The ``outline`` consisting in a list of 'steps' that you want to run, put in the right sequence.
+        This is obtained by means of the method ``spec.outline()`` which takes as input the steps.
+        *Note*: in this example we just split the main execution in two sequential steps, that is, first ``run_eos`` then ``results``.
+        However, more complex logic is allowed, as will be explained in :ref:`another section<2019_sintef_convpressure>`.
```

Comment:
Should we be referencing another tutorial here or basically copy/pasting docs/pages/2019_SINTEF/sections/appendix_workflow_logic.rst into this tutorial folder.
Duplication is never great, but the tutorials should probably be kept isolated

Reply:
I think @mbercx just forgot to change the link. I created a new label in the current pages